extension/src/fileSystem/watcher.ts

```diff
@@ -9,17 +15,26 @@ const getRelativePatternForOutsideWorkspace = (
   pattern: string
 ): RelativePattern => new RelativePattern(uri, pattern)
 
+const createRelativePattern = (
+  folder: WorkspaceFolder,
+  path: string,
+  pattern: string
+) => {
+  return new RelativePattern(
+    folder,
+    `${path}${path.length > 0 ? '/' : ''}${pattern}`
+  )
+}
+
 export const getRelativePattern = (
   path: string,
   pattern: string
 ): RelativePattern => {
   for (const workspaceFolder of workspace.workspaceFolders || []) {
     const workspaceFolderPath = workspaceFolder.uri.fsPath
     if (isSameOrChild(workspaceFolderPath, path)) {
-      return new RelativePattern(
-        workspaceFolder,
-        join(relative(workspaceFolderPath, path), pattern)
```

Comment:
Took wayyyy too long for me to realize this, but `RelativePattern` requires the `pattern` to be created with forward slashes 🤦. `join` will add backward slashes in a windows environment, thus breaking things in windows.